src/app/Main/Body/QueryPage/Parameter/QueryPage-ParameterContent.component.tsx

```diff
@@ -34,14 +34,17 @@ export function QueryPageParameterContentComponent(props: {
      */
     const createParameterComponent = (): JSX.Element[] => {
         const elements: JSX.Element[] = [];
-        Object.keys(displayFilters).forEach((element): void => {
-            elements.push(
-                <ParameterContentListComponent
-                    key={`parameter_list_${element}`}
-                    element={element}
-                    listElements={displayFilters[element]}
-                />
-            );
+        Object.keys(displayedFilters).forEach((element): void => {
```

Comment:
replace with ``for(let element of displayedFilters.keys())``. Then you don't need the undefined check. (with Object.keys you iterate over the map object with all it's internal stuff, so only use Object.keys for plain java script objects.

Reply:
"for ... of" loops are not allowed by the linter. "iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations. eslint-no-restricted-syntax"

Reply:
lol, can you please research this rule and try to explain what it means or what the actual impact is? Common sense is to prefer clean code over performance, except a part of code is a performance bottleneck.

Reply:
I haven't used it though, because I noticed that I can also get the information directly from the selectedFilters without creating a new Map. 
But from now on, I will pay more attention to whether I can use Map.

Reply:
Funny i thought of that too, but didn't blamed it because i thought there might be a reason why the selectedFilters are checked against the mainFilters.
For further reading: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Keyed_collections